ledger/src/blockstore_db.rs

```diff
@@ -955,10 +989,40 @@ pub struct WriteBatch<'a> {
     map: HashMap<&'static str, &'a ColumnFamily>,
 }
 
+pub enum ShredStorageType {
+    // Stores shreds under RocksDB's default compaction (level).
+    RocksLevel,
+    // (Experimental) Stores shreds under RocksDB's FIFO compaction which
+    // allows ledger store to reclaim storage more efficiently with
+    // lower I/O overhead.
+    RocksFifo,
+}
+
 pub struct BlockstoreOptions {
+    // The access type of blockstore. Default: PrimaryOnly
     pub access_type: AccessType,
+    // Whether to open a blockstore under a recovery mode. Default: None.
     pub recovery_mode: Option<BlockstoreRecoveryMode>,
+    // Whether to allow unlimited number of open files. Default: true.
     pub enforce_ulimit_nofile: bool,
+    // Determine how to store both data and coding shreds. Default: RocksLevel.
+    pub shred_storage_type: ShredStorageType,
```

Comment:
nit: `shred_storage_type` -> `shred_compaction_type`

Reply:
The reason behind the `storage_type` naming is because it gives us the flexibility to later switch to another storage such as the custom storage developed by @steviez.

Since the variable stores the storage type, the value for Rocks usage should be RocksLevel and RocksFIFO instead of just Level and FIFO.